### PR TITLE
[reggen] Define a RegBlock class to represent the registers, windows etc. in a Block

### PR DESCRIPTION
**Pull request by rswarbrick**
This is the next step in my on-going reggen clean-up push. There are a couple of minor refactorings in the PR, but the main change is the last commit, with the following commit message:

**[reggen] Define a class to represent the registers in a block**

This is used to represent `obj['registers']`, which was previously a
list of registers, multiregs, windows, reserved and skipto entries.
The commit doesn't replace the top-level object (obj).

Other than removing `genwennames` and `gennextoffset` from (h)json export,
the only other change is that we no longer distinguish between
`reserved` and `skipto`, which means we have to pick one format or the
other on export. This commit picks reserved (because it is more
"relocatable"), but `skipto` could work instead.

### CODE REVIEW

**Comment by rswarbrick** [comment]
This has gained a little more urgency now that I've based #5391 on it. @eunchan, @tjaychen, @msfschaffner: Would you guys mind taking a look at it?

**Comment by eunchan** [comment]
> This has gained a little more urgency now that I've based #5391 on it. @eunchan, @tjaychen, @msfschaffner: Would you guys mind taking a look at it?

Sorry. I am reviewing now.

**Review by eunchan — APPROVED** [review]
At a glance, it looks like the block now has `reg_block` (RegBlock class type) and moved validation codes into the block. The block is a list of regs, multiregs, and windows. The offset is calculated inside the block rather than outside in previous codes.

If my understand is correct, I think the code is good, more self-contained, and manageable.

**Review by msfschaffner — APPROVED** [review]
LGTM, thanks for doing this!

File: util/reggen/data.py [hunk 1]
```diff
@@ -5,9 +5,6 @@
 from collections import OrderedDict
 import re
 
-from .multi_register import MultiRegister
-from .register import Register
-
 
 # helper funtion that strips trailing _number (used as multireg suffix) from name
```
[the remark this answers is not in the document]
Comment (rswarbrick) [reply util/reggen/data.py:9]: Can do!

**Comment by rswarbrick** [comment]
Thanks for the review, @eunchan. The prod was more for Michael: I'd told him the other day that this was super-low priority but have now changed my mind :-D

**Pull request merged**